src/turtle/env/Dht.d

```diff
@@ -389,6 +389,62 @@ public class Dht : Node!(DhtNode, "dht")
         ));
     }
 
+    /***************************************************************************
+
+        Waits until the specified number of listeners are registered with the
+        specified channel.
+
+        Params:
+            channel = DHT channel to lookup
+            expected_listeners = number of listeners to wait for
+            timeout = limit of time to wait (seconds)
+            check_interval = how often to poll for changes (seconds)
+
+        Throws:
+            TestException if timeout has been hit
+
+    ***************************************************************************/
+
+    public void expectListeners ( cstring channel, uint expected_listeners,
+        double timeout = 1.0, double check_interval = 0.05 )
+    {
+        this.expectListeners((&channel)[0..1], expected_listeners, timeout,
+            check_interval);
+    }
+
+    /***************************************************************************
+
+        Waits until the specified number of listeners are registered with the
+        specified channels.
+
+        Params:
+            channels = DHT channels to lookup
+            expected_listeners = number of listeners to wait for
+            timeout = limit of time to wait (seconds)
+            check_interval = how often to poll for changes (seconds)
+
+        Throws:
+            TestException if timeout has been hit
+
+    ***************************************************************************/
+
+    public void expectListeners ( cstring[] channels, uint expected_listeners,
+        double timeout = 1.0, double check_interval = 0.05 )
+    {
```

Comment:
Why do you presume that the number of registered listeners is the same for all channels?

Reply:
It's a reasonable simplification for a common integration-test use case: you have a single app that is expected to listen to several different channels, hence the number of listeners can reasonably be assumed to match the total number of app nodes spawned.

If for some reason this assumption doesn't apply, you can use the single-channel overload on a channel-by-channel basis to deal with the custom numbers.

Reply:
So I presume the `expected_listeners` equals `1` usually, right?

Reply:
Unless one is running integration tests with multiple app nodes, yes.

Reply:
My point here is, say we have a integration test scenario with more the one app registering to > 1 channel. Then with this method it would imply that all apps have to register to the same channels, which for me somehow contradicts this level of abstraction. What I would assume then is `expected_listeners` to be a map from `channel` → `nr. of expected_listeners for this channel`.

Reply:
@matthias-wende-sociomantic Do you have an actual scenario like that?

Note that nothing stops you using the single-channel overload to handle that case.  This is just a convenience overload to deal with a very common use-case.

Reply:
To be more precise, what I'd recommend for that use-case:

* use the array-based overload to handle multiple channels where the number of listeners is expected to be the same

* use the single-channel-name overload to handle channels expected to have a unique number of listeners

I'm not anti the idea of a map-based overload, but it's not really clear what benefit that would offer over the above suggestion, and in any case I think we should have an actual use-case before implementing something like that.  We do however _definitely_ have a use-case for the existing array-based overload with a common number of listeners.

Reply:
Alright, I see that the scenario having different apps running in a turtle environment is unlikely to happen any time soon, whereas having several nodes of one application (registering all to the same channels) is a viable test cast for some apps (e.g. thruster).